From 7ce470b112469c7e63bb9aa4d68c076c0984405a Mon Sep 17 00:00:00 2001
From: Varun <vyaparla@habana.ai>
Date: Fri, 04 Mar 2022 05:03:54 +0200
Subject: [PATCH] [SW-64613] gaudi2: sec: remove unconditional rr config

This change avoids unconditional rr configuration in
ppboot. This was added initially to avoid reading fake
security boot info in ppboot but now fake security info
is read from cold reset flops. This change adds a security
check while configuring range registers in ppboot and
removes opening up of rr's in preboot when security is
not enabled

Change-Id: Ie646dac75ccc8c105a7f3accb048f6b62cd52ce4
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 1c04a19..7142e7d 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1137,30 +1137,17 @@
 void ppboot_security_init(void)
 {
 	ppboot_secure_elbi();
+
+	if (!preboot_security_is_enabled())
+		return;
+
 	ppboot_secure_lbw();
 	ppboot_secure_hbw();
 }
 
-static void preboot_unlock_pcie_rr(void)
+static void preboot_prot_bits_reset(void)
 {
-	struct hl_rr_eng_select rr_sel;
 	u8 i;
-
-	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-	/* select pcie rr */
-	rr_sel.pcie = 1;
-
-	/*
-	 * Open LBW RR that was closed in ppboot.
-	 */
-	hl_secure_lbw_rr(HL_SEC_LBW_DBG_CFG_SPACE, 0, 0, rr_sel);
-
-	/*
-	 * Open HBW RR that was closed in ppboot.
-	 */
-	hl_secure_hbw_rr(HL_SEC_HBW_ICCM0_TO_SP_SRAM, 0, 0, rr_sel);
-	hl_secure_hbw_rr(HL_SEC_HBW_SRAM, 0, 0, rr_sel);
-	hl_secure_hbw_rr(HL_SEC_HBW_SPI_FLASH, 0, 0, rr_sel);
 
 	/*
 	 * Reset protection bits configuration
@@ -1221,11 +1208,7 @@
 		preboot_secure_hbw();
 		preboot_puts("Security enabled.\n");
 	} else {
-		/*
-		 * As in ppboot we lock only pcie RR, we need to unlock the same
-		 * in preboot in case security is not enabled.
-		 */
-		preboot_unlock_pcie_rr();
+		preboot_prot_bits_reset();
 		preboot_puts("Security disabled.\n");
 	}
 }
